File: 1814_count_nice_pairs_in_an_array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countNicePairs = function(nums) {
    // function to reverse number
    function rev(k) {
        let res = 0;
        while (k != 0) {
            let t = k % 10;
            res = (res * 10) + t;
            k = Math.floor(k / 10);
        }
        return res;
    }

    let arr=[]
    for (let i of nums) {
        let key = i - rev(i);
            arr.push(key)
    }
    console.log(arr)
    let counter = 0;
    let dict={}
    let  mod = (10 ** 9)+ 7
    arr.forEach((value) => {
        if (!dict[value]) {
            dict[value] = 0;
        }
        counter = (counter + dict[value]) % mod;
        dict[value] += 1;
    });
    return counter
};

File: 1814_count_nice_pairs_in_an_array.py
class Solution:
    def countNicePairs(self, nums: List[int]) -> int:
        def rev(k):
            res = 0
            while k != 0:
                t = k % 10
                res = (res * 10) + t
                k = k // 10
            return res

        arr = []
        for i in nums:
            key = i - rev(i)
            arr.append(key)

        print(arr)

        counter = 0
        dict_ = {}
        mod = 10**9 + 7

        for value in arr:
            if value not in dict_:
                dict_[value] = 0
            counter = (counter + dict_[value]) % mod
            dict_[value] += 1

        return counter

